### PR TITLE
[clang-format] Correctly annotate braces in macros

### LINKED ISSUE #86550: clang-format 18 idempotent regression with #define within an initializer
Running clang-format 18 twice produces different results when we have #define statements within an initializer.  Running with clang-format 17.0.5 shows no difference on repeated invocations.

Given `idempotent_define.cpp`

```c++
int foo(MyType aFooSizeKind) {
    static const AType0123456789 table{
#define FOO_SIZE_KIND_2_OP_ROW(row) {GET_SYM_SIZE_KIND(row), GET_OP(row)}

        COMMA_LIST_OF(FOO_SIZE_KIND_2_OP_ROW, FOOBAR12_BINOP_SIZE_LIST),
        COMMA_LIST_OF(FOO_SIZE_KIND_2_OP_ROW, FOOBAR12_UNARY_SIZE_LIST)};

    return table.at(aFooSizeKind);
}
```

using `_clang-format`:

```yaml
---
BasedOnStyle: Google
IndentWidth: 4
ColumnLimit: 100
...
```

clang-format 18.1.1 yields the following differences on successive runs. Notice that define flips between being on a single line and multiple lines.

```
/path/to/llvm-18.1.1/bin/clang-format -style=file idempotent_define.cpp >out.18.1.1.a.cpp 2>&1
/path/to/llvm-18.1.1/bin/clang-format -style=file out.18.1.1.a.cpp >out.18.1.1.b.cpp 2>&1
/path/to/llvm-18.1.1/bin/clang-format -style=file out.18.1.1.b.cpp >out.18.1.1.c.cpp 2>&1
```

```diff
diff -u out.18.1.1.a.cpp out.18.1.1.b.cpp
--- out.18.1.1.a.cpp	2024-03-25 14:02:26.681362489 -0400
+++ out.18.1.1.b.cpp	2024-03-25 14:02:26.697362606 -0400
@@ -1,7 +1,6 @@
 int foo(MyType aFooSizeKind) {
     static const AType0123456789 table{
-#define FOO_SIZE_KIND_2_OP_ROW(row) \
-    { GET_SYM_SIZE_KIND(row), GET_OP(row) }
+#define FOO_SIZE_KIND_2_OP_ROW(row) {GET_SYM_SIZE_KIND(row), GET_OP(row)}
 
         COMMA_LIST_OF(FOO_SIZE_KIND_2_OP_ROW, FOOBAR12_BINOP_SIZE_LIST),
         COMMA_LIST_OF(FOO_SIZE_KIND_2_OP_ROW, FOOBAR12_UNARY_SIZE_LIST)};

diff -u out.18.1.1.b.cpp out.18.1.1.c.cpp
--- out.18.1.1.b.cpp	2024-03-25 14:02:26.697362606 -0400
+++ out.18.1.1.c.cpp	2024-03-25 14:02:26.715362737 -0400
@@ -1,6 +1,7 @@
 int foo(MyType aFooSizeKind) {
     static const AType0123456789 table{
-#define FOO_SIZE_KIND_2_OP_ROW(row) {GET_SYM_SIZE_KIND(row), GET_OP(row)}
+#define FOO_SIZE_KIND_2_OP_ROW(row) \
+    { GET_SYM_SIZE_KIND(row), GET_OP(row) }
 
         COMMA_LIST_OF(FOO_SIZE_KIND_2_OP_ROW, FOOBAR12_BINOP_SIZE_LIST),
         COMMA_LIST_OF(FOO_SIZE_KIND_2_OP_ROW, FOOBAR12_UNARY_SIZE_LIST)};

diff -u out.18.1.1.a.cpp out.18.1.1.c.cpp
```

### PR DESCRIPTION
**Pull request by owenca**
Also fix unit tests and reformat polly.

Fixes #86550.

### CODE REVIEW

File: clang/lib/Format/UnwrappedLineParser.cpp [hunk 1]
```diff
@@ -538,16 +538,6 @@ void UnwrappedLineParser::calculateBraceTypes(bool ExpectClassBody) {
...
         } else {
-          // Skip NextTok over preprocessor lines, otherwise we may not
-          // properly diagnose the block as a braced intializer
-          // if the comma separator appears after the pp directive.
-          while (NextTok->is(tok::hash)) {
-            ScopedMacroState MacroState(*Line, Tokens, NextTok);
-            do {
-              NextTok = Tokens->getNextToken();
-            } while (NextTok->isNot(tok::eof));
-          }
```
Comment (owenca) [inline clang/lib/Format/UnwrappedLineParser.cpp:549]: This loop is superfluous due to line 498 above.

**Comment by owenca** [comment]
The polly formatting failure should be ignored as CI is using clang-format 18.1.1 instead of the version built from this patch.

**Review by mydeveloperday — APPROVED** [review]
LGTM

**Review by HazardyKnusperkeks — APPROVED** [review]

**Comment by owenca** [comment]
@tstellar somehow this is not in 18.1.4, but the LLVM Release Status in the Projects box says "Status: Done". Did I miss something here?

**Comment by tstellar** [comment]
@owenca Is there a PR for the cherry-pick ?

**Comment by owenca** [comment]
Ah, I still need to do that.

**Pull request merged**